crates/nri-resctrl-plugin/src/lib.rs

```diff
@@ -332,6 +341,196 @@ impl<P: FsProvider> ResctrlPlugin<P> {
             self.emit_pod_add_or_update(&pod_uid, ps);
         }
     }
+
+    /// Try to create a resctrl group for a pod if currently Failed.
+    /// Emits AddOrUpdate only on state transition.
+    pub fn retry_group_creation(&self, pod_uid: &str) -> resctrl::Result<ResctrlGroupState> {
+        use std::io;
+        use std::path::PathBuf;
+
+        // Snapshot decision under lock
+        let should_try = {
+            let st = self.state.lock().unwrap();
+            match st.pods.get(pod_uid) {
+                Some(ps) => matches!(ps.group_state, ResctrlGroupState::Failed),
+                None => {
+                    return Err(resctrl::Error::Io {
+                        path: PathBuf::from("<pod>"),
+                        source: io::Error::from_raw_os_error(libc::ENOENT),
+                    })
+                }
+            }
+        };
+
+        if !should_try {
+            // Either does not exist or already Exists; return current state
+            let st = self.state.lock().unwrap();
+            let current = st.pods.get(pod_uid).unwrap().group_state.clone();
+            return Ok(current);
```

Comment:
let's combine this block into the previous block computing should_try. if the group state is does not match Failed, then return the group_state. This way we don't unlock and lock again (where things could chagnge, e.g., the pod could be removed)